diff -Naurw efivar-0.21.orig/src/linux.c efivar-0.21/src/linux.c
--- efivar-0.21.orig/src/linux.c	2015-07-13 20:00:06.000000000 +0000
+++ efivar-0.21/src/linux.c	2015-08-12 18:51:32.820404791 +0000
@@ -129,8 +129,8 @@
 		return -1;
 	}
 
-	maj = gnu_dev_major(statbuf.st_rdev);
-	min = gnu_dev_minor(statbuf.st_rdev);
+	maj = major(statbuf.st_rdev);
+	min = minor(statbuf.st_rdev);
 
 	rc = sysfs_readlink(&linkbuf, "/sys/dev/block/%u:%u", maj, min);
 	if (rc < 0)
