diff -Naurw gcc-5.2.0.orig/libsanitizer/tsan/tsan_platform_linux.cc gcc-5.2.0/libsanitizer/tsan/tsan_platform_linux.cc
--- gcc-5.2.0.orig/libsanitizer/tsan/tsan_platform_linux.cc	2014-11-13 12:41:38.000000000 -0800
+++ gcc-5.2.0/libsanitizer/tsan/tsan_platform_linux.cc	2015-10-19 20:43:54.213961010 -0700
@@ -377,7 +377,7 @@
 int ExtractResolvFDs(void *state, int *fds, int nfd) {
 #if SANITIZER_LINUX
   int cnt = 0;
-  __res_state *statp = (__res_state*)state;
+  res_state statp = (res_state)state;
   for (int i = 0; i < MAXNS && cnt < nfd; i++) {
     if (statp->_u._ext.nsaddrs[i] && statp->_u._ext.nssocks[i] != -1)
       fds[cnt++] = statp->_u._ext.nssocks[i];
