diff -Naur chromium-53.0.2785.143.orig/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc chromium-53.0.2785.143/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
--- chromium-53.0.2785.143.orig/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc	2016-10-13 09:09:41.838917314 -0700
+++ chromium-53.0.2785.143/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc	2016-10-13 09:13:06.609157786 -0700
@@ -37,7 +37,9 @@
 #endif  // defined(OS_ANDROID) && defined(__arm__)
 
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(__GLIBC__)
 #include <gnu/libc-version.h>
+#endif
 
 #include "base/version.h"
 #if defined(USE_X11)
@@ -168,7 +170,7 @@
 }
 
 void RecordLinuxGlibcVersion() {
-#if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(__GLIBC__) && !defined(OS_CHROMEOS)
   Version version(gnu_get_libc_version());
 
   UMALinuxGlibcVersion glibc_version_result = UMA_LINUX_GLIBC_NOT_PARSEABLE;
