diff -Naur chromium-55.0.2883.75.orig/base/threading/platform_thread_linux.cc chromium-55.0.2883.75/base/threading/platform_thread_linux.cc
--- chromium-55.0.2883.75.orig/base/threading/platform_thread_linux.cc	2016-12-04 17:37:29.277308691 -0800
+++ chromium-55.0.2883.75/base/threading/platform_thread_linux.cc	2016-12-04 17:44:02.811499160 -0800
@@ -156,7 +156,7 @@
 
 size_t GetDefaultThreadStackSize(const pthread_attr_t& attributes) {
 #if !defined(THREAD_SANITIZER)
-  return 0;
+  return 1 << 21;
 #else
   // ThreadSanitizer bloats the stack heavily. Evidence has been that the
   // default stack size isn't enough for some browser tests.
diff -Naur chromium-55.0.2883.75.orig/third_party/WebKit/Source/platform/heap/StackFrameDepth.cpp chromium-55.0.2883.75/third_party/WebKit/Source/platform/heap/StackFrameDepth.cpp
--- chromium-55.0.2883.75.orig/third_party/WebKit/Source/platform/heap/StackFrameDepth.cpp	2016-12-04 17:37:39.322415654 -0800
+++ chromium-55.0.2883.75/third_party/WebKit/Source/platform/heap/StackFrameDepth.cpp	2016-12-04 17:45:06.959182224 -0800
@@ -68,7 +68,7 @@
 // FIXME: On Mac OSX and Linux, this method cannot estimate stack size
 // correctly for the main thread.
 
-#if defined(__GLIBC__) || OS(ANDROID) || OS(FREEBSD)
+#if OS(LINUX) || OS(ANDROID) || OS(FREEBSD)
   // pthread_getattr_np() can fail if the thread is not invoked by
   // pthread_create() (e.g., the main thread of webkit_unit_tests).
   // If so, a conservative size estimate is returned.
@@ -135,7 +135,7 @@
 }
 
 void* StackFrameDepth::getStackStart() {
-#if defined(__GLIBC__) || OS(ANDROID) || OS(FREEBSD)
+#if OS(LINUX) || OS(ANDROID) || OS(FREEBSD)
   pthread_attr_t attr;
   int error;
 #if OS(FREEBSD)
