diff -Naur chromium-50.0.2661.37.orig/third_party/WebKit/Source/platform/heap/StackFrameDepth.cpp chromium-50.0.2661.37/third_party/WebKit/Source/platform/heap/StackFrameDepth.cpp
--- chromium-50.0.2661.37.orig/third_party/WebKit/Source/platform/heap/StackFrameDepth.cpp	2016-03-15 21:01:45.000000000 -0700
+++ chromium-50.0.2661.37/third_party/WebKit/Source/platform/heap/StackFrameDepth.cpp	2016-03-30 21:32:12.230262306 -0700
@@ -84,7 +84,7 @@
     // FIXME: On Mac OSX and Linux, this method cannot estimate stack size
     // correctly for the main thread.
 
-#if defined(__GLIBC__) || OS(ANDROID) || OS(FREEBSD)
+#if OS(LINUX) || OS(ANDROID) || OS(FREEBSD)
     // pthread_getattr_np() can fail if the thread is not invoked by
     // pthread_create() (e.g., the main thread of webkit_unit_tests).
     // If so, a conservative size estimate is returned.
@@ -146,7 +146,7 @@
 
 void* StackFrameDepth::getStackStart()
 {
-#if defined(__GLIBC__) || OS(ANDROID) || OS(FREEBSD)
+#if OS(LINUX) || OS(ANDROID) || OS(FREEBSD)
     pthread_attr_t attr;
     int error;
 #if OS(FREEBSD)
