diff -Naur chromium-54.0.2840.34.orig/base/trace_event/malloc_dump_provider.cc chromium-54.0.2840.34/base/trace_event/malloc_dump_provider.cc
--- chromium-54.0.2840.34.orig/base/trace_event/malloc_dump_provider.cc	2016-10-07 17:53:10.306270172 -0700
+++ chromium-54.0.2840.34/base/trace_event/malloc_dump_provider.cc	2016-10-07 17:56:08.523752431 -0700
@@ -224,7 +224,7 @@
   resident_size = all_heap_info.committed_size;
   allocated_objects_size = all_heap_info.allocated_size;
   allocated_objects_count = all_heap_info.block_count;
-#else
+#elif defined(OS_LINUX) && defined(__GLIBC__)
   struct mallinfo info = mallinfo();
   DCHECK_GE(info.arena + info.hblkhd, info.uordblks);
 
diff -Naur chromium-54.0.2840.34.orig/content/child/content_child_helpers.cc chromium-54.0.2840.34/content/child/content_child_helpers.cc
--- chromium-54.0.2840.34.orig/content/child/content_child_helpers.cc	2016-10-07 17:53:14.176324249 -0700
+++ chromium-54.0.2840.34/content/child/content_child_helpers.cc	2016-10-07 17:56:46.627204676 -0700
@@ -13,7 +13,7 @@
 #include "build/build_config.h"
 #include "v8/include/v8.h"
 
-#if defined(OS_LINUX)
+#if defined(OS_LINUX) && defined(__GLIBC__)
 #include <malloc.h>
 #endif
 
@@ -25,7 +25,7 @@
 // though, this provides only a partial and misleading value.
 // Unfortunately some telemetry benchmark rely on it and these need to
 // be refactored before getting rid of this. See crbug.com/581365 .
-#if defined(OS_LINUX) || defined(OS_ANDROID)
+#if (defined(OS_LINUX) && defined(__GLIBC__)) || defined(OS_ANDROID)
 size_t GetMemoryUsageKB() {
   struct mallinfo minfo = mallinfo();
   uint64_t mem_usage =
