diff -Naur chromium-53.0.2785.143.orig/base/trace_event/malloc_dump_provider.cc chromium-53.0.2785.143/base/trace_event/malloc_dump_provider.cc
--- chromium-53.0.2785.143.orig/base/trace_event/malloc_dump_provider.cc	2016-10-12 14:36:54.821809488 -0700
+++ chromium-53.0.2785.143/base/trace_event/malloc_dump_provider.cc	2016-10-12 14:40:53.295875322 -0700
@@ -128,7 +128,7 @@
   // fixed quantum, so the excess region will not be resident.
   // See crrev.com/1531463004 for detailed explanation.
   resident_size = stats.max_size_in_use;
-#else
+#elif defined(OS_LINUX) && defined(__GLIBC__)
   struct mallinfo info = mallinfo();
   DCHECK_GE(info.arena + info.hblkhd, info.uordblks);
 
@@ -140,6 +140,7 @@
   allocated_objects_size = info.uordblks;
 #endif
 
+#if defined(USE_TCMALLOC) || defined(OS_MACOSX) || defined(OS_IOS) || (defined(OS_LINUX) && defined(__GLIBC__))
   MemoryAllocatorDump* outer_dump = pmd->CreateAllocatorDump("malloc");
   outer_dump->AddScalar("virtual_size", MemoryAllocatorDump::kUnitsBytes,
                         total_virtual_size);
@@ -162,6 +163,7 @@
                           MemoryAllocatorDump::kUnitsBytes,
                           resident_size - allocated_objects_size);
   }
+#endif
 
   // Heap profiler dumps.
   if (!heap_profiler_enabled_)
diff -Naur chromium-53.0.2785.143.orig/content/child/content_child_helpers.cc chromium-53.0.2785.143/content/child/content_child_helpers.cc
--- chromium-53.0.2785.143.orig/content/child/content_child_helpers.cc	2016-10-12 14:36:57.531855693 -0700
+++ chromium-53.0.2785.143/content/child/content_child_helpers.cc	2016-10-12 14:41:55.688939085 -0700
@@ -13,7 +13,7 @@
 #include "build/build_config.h"
 #include "v8/include/v8.h"
 
-#if defined(OS_LINUX)
+#if defined(OS_LINUX) && defined(__GLIBC__)
 #include <malloc.h>
 #endif
 
@@ -25,7 +25,7 @@
 // though, this provides only a partial and misleading value.
 // Unfortunately some telemetry benchmark rely on it and these need to
 // be refactored before getting rid of this. See crbug.com/581365 .
-#if defined(OS_LINUX) || defined(OS_ANDROID)
+#if (defined(OS_LINUX) && defined(__GLIBC__)) || defined(OS_ANDROID)
 size_t GetMemoryUsageKB() {
   struct mallinfo minfo = mallinfo();
   uint64_t mem_usage =
