diff -Naur chromium-54.0.2840.34.orig/base/allocator/allocator_shim_internals.h chromium-54.0.2840.34/base/allocator/allocator_shim_internals.h
--- chromium-54.0.2840.34.orig/base/allocator/allocator_shim_internals.h	2016-10-10 09:44:17.539584695 -0700
+++ chromium-54.0.2840.34/base/allocator/allocator_shim_internals.h	2016-10-10 09:46:36.876784100 -0700
@@ -7,8 +7,6 @@
 
 #if defined(__GNUC__)
 
-#include <sys/cdefs.h>  // for __THROW
-
 #ifndef __THROW  // Not a glibc system
 #ifdef _NOEXCEPT  // LLVM libc++ uses noexcept instead
 #define __THROW _NOEXCEPT
