diff -Naurw connman-1.29.orig/configure.ac connman-1.29/configure.ac
--- connman-1.29.orig/configure.ac	2015-04-19 22:16:21.000000000 +0000
+++ connman-1.29/configure.ac	2015-08-12 16:13:18.980012215 +0000
@@ -171,6 +171,8 @@
 AM_CONDITIONAL(PPTP, test "${enable_pptp}" != "no")
 AM_CONDITIONAL(PPTP_BUILTIN, test "${enable_pptp}" = "builtin")
 
+AC_CHECK_MEMBERS([struct in6_pktinfo.ipi6_addr], [], [], [[#include <netinet/in.h>]])
+AC_CHECK_HEADERS(execinfo.h)
 AC_CHECK_HEADERS(resolv.h, dummy=yes,
 	AC_MSG_ERROR(resolver header files are required))
 AC_CHECK_LIB(resolv, ns_initparse, dummy=yes, [
diff -Naurw connman-1.29.orig/gdhcp/common.c connman-1.29/gdhcp/common.c
--- connman-1.29.orig/gdhcp/common.c	2015-04-19 22:16:21.000000000 +0000
+++ connman-1.29/gdhcp/common.c	2015-08-12 16:14:27.646680928 +0000
@@ -22,6 +22,7 @@
 #include <config.h>
 #endif
 
+#define _GNU_SOURCE
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
@@ -40,6 +41,8 @@
 #include "gdhcp.h"
 #include "common.h"
 
+#include <linux/if.h>
+
 static const DHCPOption client_options[] = {
 	{ OPTION_IP,			0x01 }, /* subnet-mask */
 	{ OPTION_IP | OPTION_LIST,	0x03 }, /* routers */
diff -Naurw connman-1.29.orig/gdhcp/common.h connman-1.29/gdhcp/common.h
--- connman-1.29.orig/gdhcp/common.h	2015-04-19 22:16:21.000000000 +0000
+++ connman-1.29/gdhcp/common.h	2015-08-12 16:20:43.420025459 +0000
@@ -18,6 +18,9 @@
  *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  *
  */
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
 
 #include <netinet/udp.h>
 #include <netinet/ip.h>
@@ -170,8 +173,8 @@
 	[OPTION_U32]	= 4,
 };
 
-/* already defined within netinet/in.h if using GNU compiler */
-#ifndef __USE_GNU
+/* already defined within netinet/in.h if using GNU compiler or musl libc */
+#ifndef HAVE_STRUCT_IN6_PKTINFO_IPI6_ADDR
 struct in6_pktinfo {
 	struct in6_addr ipi6_addr;  /* src/dst IPv6 address */
 	unsigned int ipi6_ifindex;  /* send/recv interface index */
diff -Naurw connman-1.29.orig/gweb/gresolv.c connman-1.29/gweb/gresolv.c
--- connman-1.29.orig/gweb/gresolv.c	2014-04-04 12:12:07.000000000 +0000
+++ connman-1.29/gweb/gresolv.c	2015-08-12 16:18:47.990022019 +0000
@@ -24,6 +24,7 @@
 #endif
 
 #include <errno.h>
+#include <stdio.h>
 #include <unistd.h>
 #include <stdarg.h>
 #include <string.h>
diff -Naurw connman-1.29.orig/plugins/wifi.c connman-1.29/plugins/wifi.c
--- connman-1.29.orig/plugins/wifi.c	2015-04-19 22:16:21.000000000 +0000
+++ connman-1.29/plugins/wifi.c	2015-08-12 16:16:39.330018186 +0000
@@ -30,9 +30,8 @@
 #include <string.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
-#include <linux/if_arp.h>
-#include <linux/wireless.h>
 #include <net/ethernet.h>
+#include <linux/wireless.h>
 
 #ifndef IFF_LOWER_UP
 #define IFF_LOWER_UP	0x10000
diff -Naurw connman-1.29.orig/src/ippool.c connman-1.29/src/ippool.c
--- connman-1.29.orig/src/ippool.c	2014-08-18 12:20:06.000000000 +0000
+++ connman-1.29/src/ippool.c	2015-08-12 16:19:11.013356038 +0000
@@ -28,7 +28,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
-#include <sys/errno.h>
+#include <errno.h>
 #include <sys/socket.h>
 
 #include "connman.h"
diff -Naurw connman-1.29.orig/src/iptables.c connman-1.29/src/iptables.c
--- connman-1.29.orig/src/iptables.c	2014-04-04 12:12:07.000000000 +0000
+++ connman-1.29/src/iptables.c	2015-08-12 16:19:29.380023252 +0000
@@ -28,7 +28,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
-#include <sys/errno.h>
+#include <errno.h>
 #include <sys/socket.h>
 #include <xtables.h>
 
diff -Naurw connman-1.29.orig/src/log.c connman-1.29/src/log.c
--- connman-1.29.orig/src/log.c	2014-04-04 12:12:07.000000000 +0000
+++ connman-1.29/src/log.c	2015-08-12 16:15:57.626683609 +0000
@@ -30,7 +30,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <syslog.h>
+#ifdef HAVE_EXECINFO_H
 #include <execinfo.h>
+#endif
 #include <dlfcn.h>
 
 #include "connman.h"
@@ -112,6 +114,7 @@
 
 static void print_backtrace(unsigned int offset)
 {
+#ifdef HAVE_EXECINFO_H
 	void *frames[99];
 	size_t n_ptrs;
 	unsigned int i;
@@ -210,6 +213,7 @@
 
 	close(outfd[1]);
 	close(infd[0]);
+#endif
 }
 
 static void signal_handler(int signo)
diff -Naurw connman-1.29.orig/src/ntp.c connman-1.29/src/ntp.c
--- connman-1.29.orig/src/ntp.c	2014-12-13 01:23:18.000000000 +0000
+++ connman-1.29/src/ntp.c	2015-08-12 16:17:19.636686053 +0000
@@ -180,7 +180,7 @@
 	msg.xmttime.fraction = htonl(transmit_timeval.tv_usec * 1000);
 
 	len = sendto(fd, &msg, sizeof(msg), MSG_DONTWAIT,
-						&addr, sizeof(addr));
+						(struct sockaddr *) &addr, sizeof(addr));
 	if (len < 0) {
 		connman_error("Time request for server %s failed (%d/%s)",
 			server, errno, strerror(errno));
diff -Naurw connman-1.29.orig/src/tethering.c connman-1.29/src/tethering.c
--- connman-1.29.orig/src/tethering.c	2014-08-18 12:20:06.000000000 +0000
+++ connman-1.29/src/tethering.c	2015-08-12 16:17:31.100019728 +0000
@@ -31,10 +31,8 @@
 #include <stdio.h>
 #include <sys/ioctl.h>
 #include <net/if.h>
-#include <linux/sockios.h>
 #include <string.h>
 #include <fcntl.h>
-#include <linux/if_tun.h>
 #include <netinet/in.h>
 #include <linux/if_bridge.h>
 
diff -Naurw connman-1.29.orig/tools/dhcp-test.c connman-1.29/tools/dhcp-test.c
--- connman-1.29.orig/tools/dhcp-test.c	2013-08-19 16:57:25.000000000 +0000
+++ connman-1.29/tools/dhcp-test.c	2015-08-12 16:16:11.580017358 +0000
@@ -33,7 +33,6 @@
 #include <arpa/inet.h>
 #include <net/route.h>
 #include <net/ethernet.h>
-#include <linux/if_arp.h>
 
 #include <gdhcp/gdhcp.h>
 
diff -Naurw connman-1.29.orig/tools/dnsproxy-test.c connman-1.29/tools/dnsproxy-test.c
--- connman-1.29.orig/tools/dnsproxy-test.c	2013-12-05 07:12:12.000000000 +0000
+++ connman-1.29/tools/dnsproxy-test.c	2015-08-12 16:17:50.780020314 +0000
@@ -24,6 +24,7 @@
 #endif
 
 #include <errno.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
diff -Naurw connman-1.29.orig/tools/private-network-test.c connman-1.29/tools/private-network-test.c
--- connman-1.29.orig/tools/private-network-test.c	2013-08-19 16:57:25.000000000 +0000
+++ connman-1.29/tools/private-network-test.c	2015-08-12 16:18:02.726687337 +0000
@@ -32,7 +32,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <signal.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/signalfd.h>
 #include <unistd.h>
 
diff -Naurw connman-1.29.orig/tools/tap-test.c connman-1.29/tools/tap-test.c
--- connman-1.29.orig/tools/tap-test.c	2013-12-05 07:12:12.000000000 +0000
+++ connman-1.29/tools/tap-test.c	2015-08-12 16:18:18.220021133 +0000
@@ -29,7 +29,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <string.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/ioctl.h>
 
 #include <netinet/in.h>
