diff -Naurw libbsd-0.8.0.orig/include/bsd/err.h libbsd-0.8.0/include/bsd/err.h
--- libbsd-0.8.0.orig/include/bsd/err.h	2015-09-22 22:59:34.000000000 -0700
+++ libbsd-0.8.0/include/bsd/err.h	2015-12-24 12:00:15.715019681 -0800
@@ -34,7 +34,7 @@
 #ifndef LIBBSD_ERR_H
 #define LIBBSD_ERR_H
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 
 #include <stdarg.h>
 
diff -Naurw libbsd-0.8.0.orig/include/bsd/libutil.h libbsd-0.8.0/include/bsd/libutil.h
--- libbsd-0.8.0.orig/include/bsd/libutil.h	2015-09-22 10:56:09.000000000 -0700
+++ libbsd-0.8.0/include/bsd/libutil.h	2015-12-24 12:00:15.718353014 -0800
@@ -40,7 +40,7 @@
 #define _LIBUTIL_H_
 
 #include <features.h>
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 #include <sys/types.h>
 #include <stdint.h>
 #include <stdio.h>
diff -Naurw libbsd-0.8.0.orig/include/bsd/md5.h libbsd-0.8.0/include/bsd/md5.h
--- libbsd-0.8.0.orig/include/bsd/md5.h	2015-09-22 22:14:21.000000000 -0700
+++ libbsd-0.8.0/include/bsd/md5.h	2015-12-24 12:00:15.721686347 -0800
@@ -27,7 +27,7 @@
 	uint8_t buffer[MD5_BLOCK_LENGTH];	/* input buffer */
 } MD5_CTX;
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 
 __BEGIN_DECLS
 void	 MD5Init(MD5_CTX *);
diff -Naurw libbsd-0.8.0.orig/include/bsd/nlist.h libbsd-0.8.0/include/bsd/nlist.h
--- libbsd-0.8.0.orig/include/bsd/nlist.h	2015-09-22 10:56:09.000000000 -0700
+++ libbsd-0.8.0/include/bsd/nlist.h	2015-12-24 12:00:15.725019681 -0800
@@ -27,8 +27,12 @@
 #ifndef LIBBSD_NLIST_H
 #define LIBBSD_NLIST_H
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
+#if defined(__GLIBC__)
 #include <a.out.h>
+#else
+#include <linux/a.out.h>
+#endif
 
 __BEGIN_DECLS
 extern int nlist(const char *filename, struct nlist *list);
diff -Naurw libbsd-0.8.0.orig/include/bsd/readpassphrase.h libbsd-0.8.0/include/bsd/readpassphrase.h
--- libbsd-0.8.0.orig/include/bsd/readpassphrase.h	2011-07-05 11:58:44.000000000 -0700
+++ libbsd-0.8.0/include/bsd/readpassphrase.h	2015-12-24 12:00:15.725019681 -0800
@@ -31,7 +31,7 @@
 #define RPP_SEVENBIT    0x10		/* Strip the high bit from input. */
 #define RPP_STDIN       0x20		/* Read from stdin, not /dev/tty */
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 #include <sys/types.h>
 
 __BEGIN_DECLS
diff -Naurw libbsd-0.8.0.orig/include/bsd/stdio.h libbsd-0.8.0/include/bsd/stdio.h
--- libbsd-0.8.0.orig/include/bsd/stdio.h	2015-09-22 22:59:34.000000000 -0700
+++ libbsd-0.8.0/include/bsd/stdio.h	2015-12-24 12:00:15.725019681 -0800
@@ -38,7 +38,7 @@
 #ifndef LIBBSD_STDIO_H
 #define LIBBSD_STDIO_H
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 #include <sys/types.h>
 
 __BEGIN_DECLS
diff -Naurw libbsd-0.8.0.orig/include/bsd/stdlib.h libbsd-0.8.0/include/bsd/stdlib.h
--- libbsd-0.8.0.orig/include/bsd/stdlib.h	2015-09-22 22:14:21.000000000 -0700
+++ libbsd-0.8.0/include/bsd/stdlib.h	2015-12-24 12:00:15.728353014 -0800
@@ -42,7 +42,7 @@
 #ifndef LIBBSD_STDLIB_H
 #define LIBBSD_STDLIB_H
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 #include <sys/stat.h>
 #include <stdint.h>
 
diff -Naurw libbsd-0.8.0.orig/include/bsd/string.h libbsd-0.8.0/include/bsd/string.h
--- libbsd-0.8.0.orig/include/bsd/string.h	2015-09-23 20:30:23.000000000 -0700
+++ libbsd-0.8.0/include/bsd/string.h	2015-12-24 12:00:15.731686348 -0800
@@ -33,7 +33,7 @@
 #ifndef LIBBSD_STRING_H
 #define LIBBSD_STRING_H
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 #include <sys/types.h>
 
 __BEGIN_DECLS
diff -Naurw libbsd-0.8.0.orig/include/bsd/stringlist.h libbsd-0.8.0/include/bsd/stringlist.h
--- libbsd-0.8.0.orig/include/bsd/stringlist.h	2015-09-22 10:56:09.000000000 -0700
+++ libbsd-0.8.0/include/bsd/stringlist.h	2015-12-24 12:00:15.731686348 -0800
@@ -31,7 +31,7 @@
 
 #ifndef _STRINGLIST_H
 #define _STRINGLIST_H
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 #include <sys/types.h>
 
 /*
diff -Naurw libbsd-0.8.0.orig/include/bsd/sys/cdefs.h libbsd-0.8.0/include/bsd/sys/cdefs.h
--- libbsd-0.8.0.orig/include/bsd/sys/cdefs.h	2015-09-22 22:59:27.000000000 -0700
+++ libbsd-0.8.0/include/bsd/sys/cdefs.h	2015-12-24 12:13:14.608376224 -0800
@@ -24,10 +24,14 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifdef LIBBSD_OVERLAY
-#include_next <sys/cdefs.h>
-#else
+#if defined(__GLIBC__)
 #include <sys/cdefs.h>
+#elif defined(__cplusplus)
+#define __BEGIN_DECLS extern "C" {
+#define __END_DECLS }
+#else
+#define __BEGIN_DECLS
+#define __END_DECLS
 #endif
 
 #ifndef LIBBSD_SYS_CDEFS_H
diff -Naurw libbsd-0.8.0.orig/include/bsd/timeconv.h libbsd-0.8.0/include/bsd/timeconv.h
--- libbsd-0.8.0.orig/include/bsd/timeconv.h	2015-09-22 10:56:09.000000000 -0700
+++ libbsd-0.8.0/include/bsd/timeconv.h	2015-12-24 12:00:15.735019681 -0800
@@ -41,7 +41,7 @@
 #ifndef _TIMECONV_H_
 #define _TIMECONV_H_
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 #include <stdint.h>
 #include <time.h>
 
diff -Naurw libbsd-0.8.0.orig/include/bsd/unistd.h libbsd-0.8.0/include/bsd/unistd.h
--- libbsd-0.8.0.orig/include/bsd/unistd.h	2015-11-28 08:22:08.000000000 -0800
+++ libbsd-0.8.0/include/bsd/unistd.h	2015-12-24 12:00:15.735019681 -0800
@@ -34,7 +34,7 @@
 #ifndef LIBBSD_UNISTD_H
 #define LIBBSD_UNISTD_H
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 #include <sys/stat.h>
 
 #ifndef S_ISTXT
diff -Naurw libbsd-0.8.0.orig/include/bsd/vis.h libbsd-0.8.0/include/bsd/vis.h
--- libbsd-0.8.0.orig/include/bsd/vis.h	2011-07-05 11:58:44.000000000 -0700
+++ libbsd-0.8.0/include/bsd/vis.h	2015-12-24 12:00:15.735019681 -0800
@@ -72,7 +72,7 @@
  */
 #define	UNVIS_END	1	/* no more characters */
 
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 
 __BEGIN_DECLS
 char	*vis(char *, int, int, int);
diff -Naurw libbsd-0.8.0.orig/include/bsd/wchar.h libbsd-0.8.0/include/bsd/wchar.h
--- libbsd-0.8.0.orig/include/bsd/wchar.h	2015-09-22 10:56:09.000000000 -0700
+++ libbsd-0.8.0/include/bsd/wchar.h	2015-12-24 12:00:15.738353015 -0800
@@ -40,7 +40,7 @@
 #define LIBBSD_WCHAR_H
 
 #include <stddef.h>
-#include <sys/cdefs.h>
+#include "sys/cdefs.h"
 #include <sys/types.h>
 
 __BEGIN_DECLS
diff -Naurw libbsd-0.8.0.orig/src/flopen.c libbsd-0.8.0/src/flopen.c
--- libbsd-0.8.0.orig/src/flopen.c	2011-07-05 09:56:44.000000000 -0700
+++ libbsd-0.8.0/src/flopen.c	2015-12-24 11:52:49.921673064 -0800
@@ -32,6 +32,7 @@
 #include <sys/stat.h>
 
 #include <errno.h>
+#include <fcntl.h>
 #include <stdarg.h>
 #include <unistd.h>
 #include <libutil.h>
diff -Naurw libbsd-0.8.0.orig/src/funopen.c libbsd-0.8.0/src/funopen.c
--- libbsd-0.8.0.orig/src/funopen.c	2015-09-22 10:56:09.000000000 -0700
+++ libbsd-0.8.0/src/funopen.c	2015-12-24 11:52:49.921673064 -0800
@@ -137,6 +137,4 @@
 
 	return fopencookie(cookiewrap, mode, funcswrap);
 }
-#else
-#error "Function funopen() needs to be ported."
 #endif
diff -Naurw libbsd-0.8.0.orig/src/nlist.c libbsd-0.8.0/src/nlist.c
--- libbsd-0.8.0.orig/src/nlist.c	2015-09-22 22:59:34.000000000 -0700
+++ libbsd-0.8.0/src/nlist.c	2015-12-24 11:52:49.921673064 -0800
@@ -39,11 +39,13 @@
 #include <arpa/inet.h>
 
 #include <errno.h>
-#include <a.out.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 
+#include "nlist.h"
+
 #if !defined(__NO_A_OUT_SUPPORT)
 #define _NLIST_DO_AOUT
 #endif
