diff -Naurw btrfs-progs-v4.2.orig/bitops.h btrfs-progs-v4.2/bitops.h
--- btrfs-progs-v4.2.orig/bitops.h	2015-09-03 05:29:20.000000000 -0700
+++ btrfs-progs-v4.2/bitops.h	2015-09-11 12:32:24.535070590 -0700
@@ -69,6 +69,9 @@
 }
 
 #define BITOP_WORD(nr)		((nr) / BITS_PER_LONG)
+#ifndef __always_inline
+#define __always_inline         inline __attribute__((always_inline))
+#endif
 
 /**
  * __ffs - find first bit in word.
diff -Naurw btrfs-progs-v4.2.orig/chunk-recover.c btrfs-progs-v4.2/chunk-recover.c
--- btrfs-progs-v4.2.orig/chunk-recover.c	2015-09-03 05:29:20.000000000 -0700
+++ btrfs-progs-v4.2/chunk-recover.c	2015-09-11 12:36:03.528410449 -0700
@@ -883,12 +883,7 @@
 		for (i = 0; i < devidx; i++) {
 			if (dev_scans[i].bytenr == -1)
 				continue;
-			ret = pthread_tryjoin_np(t_scans[i],
-						 (void **)&t_rets[i]);
-			if (ret == EBUSY) {
-				all_done = 0;
-				continue;
-			}
+			ret = pthread_join(t_scans[i], (void **)&t_rets[i]);
 			if (ret || t_rets[i]) {
 				ret = 1;
 				goto out1;
diff -Naurw btrfs-progs-v4.2.orig/kerncompat.h btrfs-progs-v4.2/kerncompat.h
--- btrfs-progs-v4.2.orig/kerncompat.h	2015-09-03 05:29:20.000000000 -0700
+++ btrfs-progs-v4.2/kerncompat.h	2015-09-11 12:32:24.538403923 -0700
@@ -28,6 +28,7 @@
 #include <assert.h>
 #include <stddef.h>
 #include <linux/types.h>
+#include <linux/limits.h>
 #include <stdint.h>
 
 #include <features.h>
diff -Naurw btrfs-progs-v4.2.orig/rbtree.h btrfs-progs-v4.2/rbtree.h
--- btrfs-progs-v4.2.orig/rbtree.h	2015-09-03 05:29:20.000000000 -0700
+++ btrfs-progs-v4.2/rbtree.h	2015-09-11 12:32:24.538403923 -0700
@@ -38,6 +38,10 @@
 extern "C" {
 #endif
 
+#ifndef __always_inline
+#define __always_inline         inline __attribute__((always_inline))
+#endif
+
 struct rb_node {
 	unsigned long  __rb_parent_color;
 	struct rb_node *rb_right;
diff -Naurw btrfs-progs-v4.2.orig/utils.c btrfs-progs-v4.2/utils.c
--- btrfs-progs-v4.2.orig/utils.c	2015-09-03 05:29:20.000000000 -0700
+++ btrfs-progs-v4.2/utils.c	2015-09-11 12:32:21.331737161 -0700
@@ -1176,13 +1176,19 @@
 {
 	int ret;
 	FILE *f;
+	struct stat stat_buf;
 	char fmt[20];
 	char p[PATH_MAX];
 	char real_loop_dev[PATH_MAX];
 
 	if (!realpath(loop_dev, real_loop_dev))
 		return -errno;
-	snprintf(p, PATH_MAX, "/sys/block/%s/loop/backing_file", strrchr(real_loop_dev, '/'));
+	
+        if (stat(real_loop_dev, &stat_buf) || !S_ISBLK(stat_buf.st_mode))
+                return -errno;
+
+        snprintf(p, PATH_MAX, "/sys/dev/block/%d:%d/loop/backing_file",
+                major(stat_buf.st_rdev), minor(stat_buf.st_rdev));
 	if (!(f = fopen(p, "r")))
 		return -errno;
 
