diff -Naurw xorg-server-1.17.4.orig/hw/kdrive/linux/linux.c xorg-server-1.17.4/hw/kdrive/linux/linux.c
--- xorg-server-1.17.4.orig/hw/kdrive/linux/linux.c	2015-10-21 11:02:28.000000000 -0700
+++ xorg-server-1.17.4/hw/kdrive/linux/linux.c	2015-10-29 10:05:48.270447080 -0700
@@ -66,8 +66,8 @@
 LinuxCheckChown(const char *file)
 {
     struct stat st;
-    __uid_t u;
-    __gid_t g;
+    uid_t u;
+    gid_t g;
     int r;
 
     if (stat(file, &st) < 0)
diff -Naurw xorg-server-1.17.4.orig/hw/xfree86/common/compiler.h xorg-server-1.17.4/hw/xfree86/common/compiler.h
--- xorg-server-1.17.4.orig/hw/xfree86/common/compiler.h	2015-10-27 13:12:00.000000000 -0700
+++ xorg-server-1.17.4/hw/xfree86/common/compiler.h	2015-10-29 10:05:48.270447080 -0700
@@ -518,7 +518,7 @@
     barrier();
 }
 
-#elif defined(__mips__) || (defined(__arm32__) && !defined(__linux__))
+#elif defined(__mips__) || (defined(__arm32__) && !defined(__GLIBC__))
 #if defined(__arm32__) || defined(__mips64)
 #define PORT_SIZE long
 #else
@@ -758,7 +758,7 @@
     return xf86ReadMmio32Le((void *) ioBase, port);
 }
 
-#elif defined(__arm__) && defined(__linux__)
+#elif defined(__arm__) && defined(__GLIBC__)
 
 /* for Linux on ARM, we use the LIBC inx/outx routines */
 /* note that the appropriate setup via "ioperm" needs to be done */
