diff -Naurw catdoc-0.94.4.orig/src/xlsparse.c catdoc-0.94.4/src/xlsparse.c
--- catdoc-0.94.4.orig/src/xlsparse.c	2012-12-03 10:03:52.000000000 -0800
+++ catdoc-0.94.4/src/xlsparse.c	2015-09-12 15:13:00.375933435 -0700
@@ -718,7 +718,7 @@
 
 	if ( *(rec) & 0x02 )
 	{
-		int32_t ival=getlong((unsigned char *)rec,0);
+		long int ival=getlong((unsigned char *)rec,0);
 		value=(double)(ival>>2);
 	}
 	else { 
