diff -Naurw pax-utils-1.1.4.orig/security.c pax-utils-1.1.4/security.c
--- pax-utils-1.1.4.orig/security.c	2015-10-25 21:22:28.000000000 -0700
+++ pax-utils-1.1.4/security.c	2016-02-08 15:58:37.216432603 -0800
@@ -72,6 +72,7 @@
 	int base_syscalls[] = {
 		/* We write the most w/scanelf.  */
 		SCMP_SYS(write),
+                SCMP_SYS(writev),
 
 		/* Then the stat family of functions.  */
 		SCMP_SYS(newfstatat),
@@ -102,6 +103,7 @@
 		/* Then the file reading functions.  */
 		SCMP_SYS(pread64),
 		SCMP_SYS(read),
+		SCMP_SYS(readv),
 
 		/* Then the fd manipulation functions.  */
 		SCMP_SYS(fcntl),
