diff -Naurw libdrm-2.4.65.orig/xf86drm.c libdrm-2.4.65/xf86drm.c
--- libdrm-2.4.65.orig/xf86drm.c	2015-09-04 05:40:57.000000000 -0700
+++ libdrm-2.4.65/xf86drm.c	2015-09-17 10:47:08.590046933 -0700
@@ -56,6 +56,9 @@
 # include <sys/mkdev.h> /* defines major(), minor(), and makedev() on Solaris */
 #endif
 #include <math.h>
+#ifdef __linux__
+#include <linux/limits.h>
+#endif
 
 /* Not all systems have MAP_FAILED defined */
 #ifndef MAP_FAILED
