diff -Naurw mpd-0.19.11.orig/src/notify.hxx mpd-0.19.11/src/notify.hxx
--- mpd-0.19.11.orig/src/notify.hxx	2016-01-02 20:41:56.861943675 -0800
+++ mpd-0.19.11/src/notify.hxx	2016-01-02 20:42:29.627943088 -0800
@@ -28,7 +28,7 @@
 	Cond cond;
 	bool pending;
 
-#if !defined(WIN32) && !defined(__NetBSD__) && !defined(__BIONIC__)
+#if defined(__GLIBC__)
 	constexpr
 #endif
 	notify():pending(false) {}
diff -Naurw mpd-0.19.11.orig/src/thread/PosixCond.hxx mpd-0.19.11/src/thread/PosixCond.hxx
--- mpd-0.19.11.orig/src/thread/PosixCond.hxx	2016-01-02 20:41:56.859943675 -0800
+++ mpd-0.19.11/src/thread/PosixCond.hxx	2016-01-02 20:43:13.209942308 -0800
@@ -41,7 +41,7 @@
 	pthread_cond_t cond;
 
 public:
-#if defined(__NetBSD__) || defined(__BIONIC__)
+#if !defined(__GLIBC__)
 	/* NetBSD's PTHREAD_COND_INITIALIZER is not compatible with
 	   "constexpr" */
 	PosixCond() {
diff -Naurw mpd-0.19.11.orig/src/thread/PosixMutex.hxx mpd-0.19.11/src/thread/PosixMutex.hxx
--- mpd-0.19.11.orig/src/thread/PosixMutex.hxx	2016-01-02 20:41:56.858943675 -0800
+++ mpd-0.19.11/src/thread/PosixMutex.hxx	2016-01-02 20:43:41.934941793 -0800
@@ -41,7 +41,7 @@
 	pthread_mutex_t mutex;
 
 public:
-#if defined(__NetBSD__) || defined(__BIONIC__)
+#if !defined(__GLIBC__)
 	/* NetBSD's PTHREAD_MUTEX_INITIALIZER is not compatible with
 	   "constexpr" */
 	PosixMutex() {
