diff -Naur plymouth-0.9.2.orig/src/libply/ply-command-parser.c plymouth-0.9.2/src/libply/ply-command-parser.c
--- plymouth-0.9.2.orig/src/libply/ply-command-parser.c	2015-02-26 10:32:41.000000000 -0800
+++ plymouth-0.9.2/src/libply/ply-command-parser.c	2016-03-04 00:03:45.193553427 -0800
@@ -76,6 +76,19 @@
         uint32_t          dispatch_is_queued : 1;
 };
 
+#ifndef rpmatch
+static int
+rpmatch (const char                              *response)
+{
+        if (strcmp(response, "yes") == 0)
+                return 1;
+        else if (strcmp(response, "no") == 0)
+                return 0;
+        else
+                return -1;
+}
+#endif
+
 static ply_command_option_t *
 ply_command_option_new (const char               *name,
                         const char               *description,
