diff -Naurw mozilla-release.orig/js/src/jit/ExecutableAllocator.cpp mozilla-release/js/src/jit/ExecutableAllocator.cpp
--- mozilla-release.orig/js/src/jit/ExecutableAllocator.cpp	2015-10-14 14:41:41.000000000 -0700
+++ mozilla-release/js/src/jit/ExecutableAllocator.cpp	2015-10-22 09:13:15.440059937 -0700
@@ -88,8 +88,4 @@
     }
 }
 
-#if TARGET_OS_IPHONE
 bool ExecutableAllocator::nonWritableJitCode = true;
-#else
-bool ExecutableAllocator::nonWritableJitCode = false;
-#endif
