diff -Naurw firefox-42.0.orig/tools/profiler/core/platform-linux.cc firefox-42.0/tools/profiler/core/platform-linux.cc
--- firefox-42.0.orig/tools/profiler/core/platform-linux.cc	2015-10-29 15:18:01.000000000 -0700
+++ firefox-42.0/tools/profiler/core/platform-linux.cc	2015-11-04 12:37:39.610069953 -0800
@@ -711,10 +711,10 @@
 {
   MOZ_ASSERT(aContext);
   ucontext_t* pContext = reinterpret_cast<ucontext_t*>(aContext);
-  if (!getcontext(pContext)) {
+/*  if (!getcontext(pContext)) {
     context = pContext;
     SetSampleContext(this, aContext);
-  }
+  } */
 }
 
 void OS::SleepMicro(int microseconds)
diff -Naurw firefox-42.0.orig/tools/profiler/core/platform.h firefox-42.0/tools/profiler/core/platform.h
--- firefox-42.0.orig/tools/profiler/core/platform.h	2015-10-29 15:18:01.000000000 -0700
+++ firefox-42.0/tools/profiler/core/platform.h	2015-11-04 12:36:40.706734864 -0800
@@ -60,7 +60,7 @@
 
 // We need a definition of gettid(), but glibc doesn't provide a
 // wrapper for it.
-#if defined(__GLIBC__)
+#if 1
 #include <unistd.h>
 #include <sys/syscall.h>
 static inline pid_t gettid()
