diff -Naurw firefox-42.0.orig/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.cc firefox-42.0/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.cc
--- firefox-42.0.orig/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.cc	2015-11-27 11:49:14.436842904 -0800
+++ firefox-42.0/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.cc	2015-11-27 11:50:29.870178485 -0800
@@ -34,7 +34,6 @@
 #include "common/stabs_reader.h"
 
 #include <assert.h>
-#include <stab.h>
 #include <string.h>
 
 #include <string>
diff -Naurw firefox-42.0.orig/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.h firefox-42.0/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.h
--- firefox-42.0.orig/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.h	2015-11-27 11:49:14.436842904 -0800
+++ firefox-42.0/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.h	2015-11-27 11:52:50.146849332 -0800
@@ -61,6 +61,17 @@
 #include <mach-o/nlist.h>
 #endif
 
+// Definitions from <stab.h> and <a.out.h> for systems which
+// do not have them
+#undef N_UNDF
+#define N_UNDF  0x0
+#define N_FUN   0x24
+#define N_SLINE 0x44
+#define N_SO    0x64
+#define N_LSYM  0x80
+#define N_BINCL 0x82
+#define N_SOL   0x84
+
 #include <string>
 #include <vector>
 
diff -Naurw firefox-42.0.orig/toolkit/crashreporter/google-breakpad/src/common/stabs_reader_unittest.cc firefox-42.0/toolkit/crashreporter/google-breakpad/src/common/stabs_reader_unittest.cc
--- firefox-42.0.orig/toolkit/crashreporter/google-breakpad/src/common/stabs_reader_unittest.cc	2015-11-27 11:49:14.436842904 -0800
+++ firefox-42.0/toolkit/crashreporter/google-breakpad/src/common/stabs_reader_unittest.cc	2015-11-27 11:50:35.980178668 -0800
@@ -33,7 +33,6 @@
 
 #include <assert.h>
 #include <errno.h>
-#include <stab.h>
 #include <stdarg.h>
 #include <stdlib.h>
 #include <string.h>
