diff -Naurw firefox-42.0.orig/toolkit/mozapps/update/common/updatedefines.h firefox-42.0/toolkit/mozapps/update/common/updatedefines.h
--- firefox-42.0.orig/toolkit/mozapps/update/common/updatedefines.h	2015-10-29 15:18:00.000000000 -0700
+++ firefox-42.0/toolkit/mozapps/update/common/updatedefines.h	2015-11-04 12:29:43.666722437 -0800
@@ -109,7 +109,7 @@
 # include <sys/wait.h>
 # include <unistd.h>
 
-#ifdef SOLARIS
+#if 1
 # include <sys/stat.h>
 #else
 # include <fts.h>
diff -Naurw firefox-42.0.orig/toolkit/mozapps/update/updater/updater.cpp firefox-42.0/toolkit/mozapps/update/updater/updater.cpp
--- firefox-42.0.orig/toolkit/mozapps/update/updater/updater.cpp	2015-10-29 15:18:00.000000000 -0700
+++ firefox-42.0/toolkit/mozapps/update/updater/updater.cpp	2015-11-04 12:29:43.670055771 -0800
@@ -3592,7 +3592,8 @@
   return rv;
 }
 
-#elif defined(SOLARIS)
+#elif 1
+//#defined(SOLARIS)
 int add_dir_entries(const NS_tchar *dirpath, ActionList *list)
 {
   int rv = OK;
@@ -3600,7 +3601,7 @@
   NS_tchar foundpath[MAXPATHLEN];
   struct {
     dirent dent_buffer;
-    char chars[MAXNAMLEN];
+    char chars[NAME_MAX];
   } ent_buf;
   struct dirent* ent;
 
