diff -Naur dhcpcd-6.10.3.orig/compat/pidfile.c dhcpcd-6.10.3/compat/pidfile.c
--- dhcpcd-6.10.3.orig/compat/pidfile.c	2016-04-20 01:37:23.000000000 -0700
+++ dhcpcd-6.10.3/compat/pidfile.c	2016-05-08 10:34:40.134142016 -0700
@@ -29,7 +29,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 #if defined(LIBC_SCCS) && !defined(lint)
 __RCSID("$NetBSD: pidfile.c,v 1.14 2016/04/12 20:40:43 roy Exp $");
 #endif
