diff -Naurw icedtea-2.6.1.orig/Makefile.am icedtea-2.6.1/Makefile.am
--- icedtea-2.6.1.orig/Makefile.am	2015-07-21 03:37:00.699650653 +0700
+++ icedtea-2.6.1/Makefile.am	2015-08-26 20:41:24.611614947 +0700
@@ -3036,9 +3036,11 @@
 	for files in $(BUILD_SDK_DIR)/jre/lib/ext/*.jar; do \
 	  $(call install_file,$${files},$(DESTDIR)${prefix}/jre/lib/ext,$(INSTALL_DATA));\
 	done
+if ENABLE_SYSTEMTAP
 	for files in $(BUILD_SDK_DIR)/tapset/*.stp; do \
 	  $(call install_file,$${files},$(DESTDIR)${prefix}/tapset,$(INSTALL_DATA));\
 	done
+endif
 	$(abs_top_builddir)/pax-mark-vm $(DESTDIR)${prefix}
 
 install-data-local:
diff -Naurw icedtea-2.6.1.orig/Makefile.in icedtea-2.6.1/Makefile.in
--- icedtea-2.6.1.orig/Makefile.in	2015-07-21 03:37:15.251397004 +0700
+++ icedtea-2.6.1/Makefile.in	2015-08-26 20:42:33.301616993 +0700
@@ -3309,9 +3309,9 @@
 	for files in $(BUILD_SDK_DIR)/jre/lib/ext/*.jar; do \
 	  $(call install_file,$${files},$(DESTDIR)${prefix}/jre/lib/ext,$(INSTALL_DATA));\
 	done
-	for files in $(BUILD_SDK_DIR)/tapset/*.stp; do \
-	  $(call install_file,$${files},$(DESTDIR)${prefix}/tapset,$(INSTALL_DATA));\
-	done
+@ENABLE_SYSTEMTAP_TRUE@	for files in $(BUILD_SDK_DIR)/tapset/*.stp; do \
+@ENABLE_SYSTEMTAP_TRUE@	  $(call install_file,$${files},$(DESTDIR)${prefix}/tapset,$(INSTALL_DATA));\
+@ENABLE_SYSTEMTAP_TRUE@	done
 	$(abs_top_builddir)/pax-mark-vm $(DESTDIR)${prefix}
 
 install-data-local:
